src/timepicker/timepicker.js

```diff
@@ -55,6 +55,50 @@ angular.module('ui.bootstrap.timepicker', [])
     });
   }
 
+  var min;
+  if ($attrs.min) {
+    $scope.$parent.$watch($parse($attrs.min), function(value) {
+      var dt = new Date(value);
+      min = isNaN(dt) ? undefined : dt;
+    });
+  }
+
+  var max;
+  if ($attrs.max) {
+    $scope.$parent.$watch($parse($attrs.max), function(value) {
+      var dt = new Date(value);
+      max = isNaN(dt) ? undefined : dt;
+    });
+  }
+
+  $scope.noIncrementHours = function() {
+    var incrementedSelected = addMinutes( selected, hourStep * 60 );
+    return incrementedSelected > max || (min && incrementedSelected < selected);
```

Comment:
I'm not sure I understand.  I was following the pattern of the other attributes, such as $attrs.hourStep and $attrs.minuteStep.  If the attribute is present, add a watch for the expression in it.